src/main/comments.js

```diff
@@ -583,10 +578,20 @@ function printComments(path, printed, options, needsSemi) {
     }
   }, "comments");
 
-  return prependCursorPlaceholder(path, options, [
-    ...leadingParts,
-    ...trailingParts,
-  ]);
+  if (isCursorNode) {
+    leadingParts.unshift(cursor);
+    trailingParts.push(cursor);
+  }
+
+  return { leading: leadingParts, trailing: trailingParts };
+}
+
+function printComments(path, doc, options) {
+  const { leading, trailing } = printCommentsSeparately(path, options);
+  if (!leading && !trailing) {
+    return doc;
+  }
+  return [leading || "", doc, trailing || ""];
```

Comment:
`leading` and `trailing` can only be `cursor`, `Array`, and `""`? So we should use `leading` directly instead of `leading || ""`.